test/unit/buffer_spec.lua

```diff
@@ -211,93 +212,177 @@ describe('buffer functions', function()
+      '%=abcde',             '~~~~~abcde', {fillchar=('~'):byte()})
+    statusline_test('should put fillchar `!` in between text', 10, 10, 10,
+      'abc%=def',            'abc!!!!def', {fillchar=('!'):byte()})
+    statusline_test('should put fillchar `~` in between text', 10, 10, 10,
+      'abc%=def',            'abc~~~~def', {fillchar=('~'):byte()})
+
+    statusline_test('should copy plain text', 80, 14, 14, 'this is a test', 'this is a test')
+    statusline_test('should print the tail file name', 80, 15, 15, '%t', 'buffer_spec.lua', {file_name='test/unit/buffer_spec.lua'})
+    statusline_test('should print the buffer number', 80, 1, 1, '%n', '1')
+    statusline_test('should print the current line number in the buffer', 80, 1, 1, '%l', '0')
+    statusline_test('should print the number of lines in the buffer', 80, 1, 1, '%L', '1')
+
+    statusline_test('should truncate when standard text pattern is too long', 10, 10, 10,
+      '0123456789abcde',     '<6789abcde')
+    statusline_test('should truncate when using =', 10, 10, 10,
+      'abcdef%=ghijkl',      'abcdef<jkl')
+    statusline_test('should truncate centered text when using ==', 10, 10, 10,
+      'abcde%=gone%=fghij',  'abcde<ghij')
+
+    statusline_test('should right align when using =', 20, 20, 20,
+      'neo%=vim',            'neo              vim')
+    statusline_test('should, when possible, center text when using %=text%=', 20, 20, 20,
+      'abc%=neovim%=def',    'abc    neovim    def')
+    statusline_test('should handle uneven spacing in the buffer when using %=text%=', 20, 20, 20,
+      'abc%=neo_vim%=def',   'abc   neo_vim    def')
+    statusline_test('should have equal spaces even with non-equal sides when using =', 20, 20, 20,
+      'foobar%=test%=baz',   'foobar   test    baz')
+    statusline_test('should have equal spaces even with longer right side when using =', 20, 20, 20,
+      'a%=test%=longtext',   'a   test    longtext')
+    statusline_test('should handle an empty left side when using ==', 20, 20, 20,
+      '%=test%=baz',         '      test       baz')
+    statusline_test('should handle an empty right side when using ==', 20, 20, 20,
+      'foobar%=test%=',      'foobar     test     ')
+    statusline_test('should handle consecutive empty ==', 20, 20, 20,
+      '%=%=test%=',          '          test      ')
+    statusline_test('should handle an = alone', 20, 20, 20,
+      '%=',                  '                    ')
+    statusline_test('should right align text when it is alone with =', 20, 20, 20,
+      '%=foo',               '                 foo')
+    statusline_test('should left align text when it is alone with =', 20, 20, 20,
+      'foo%=',               'foo                 ')
+    statusline_test('should handle a much larger amount of = than buffer locations', 20, 20, 20,
+      ('%='):rep(STL_MAX_ITEM), '                    ') -- Should be fine, because within limit
```

Comment:
Just interested: what is it going to do with `'a' .. ('%=a'):rep(STL_MAX_ITEM)`. Also with different locations of `%<`. Though I do not see much tests which involve truncation here: only three basic tests and no `%<`. Especially no multiple `%<`.